### PR TITLE
Overhaul styling (fonts, colors, spacing, layout)

### PR DESCRIPTION
**Pull request by tjennison-google**
Some notable changes in the code:
* Added an MUI theme to control the overall style. This is a much cleaner interface for customizing the MUI components than using CSS. Future work will remove all built in style from app.css and set up a method for plugins to override the theme.
* Fixed ellipses in the TreeGrid.
* Tightened some of the column sizes to better accomodate longer concept names and reasonably support 1280 pixel width displays.
* Improve horizontal scrolling of tables by making the scroll bar stick to the bottom of the screen instead of the bottom of the table which might be way below the bottom of the screen.
* Removed existing inline concept editing. It was unnecessary with the change to single select. In the future, this is where modifier controls will go.

### CODE REVIEW

**Comment by tjennison-google** [comment]
image
image
image

File: ui/src/datasets.tsx [hunk 1]
```diff
@@ -133,6 +133,10 @@ export function Datasets() {
   });
 
   const allAttributesChecked = () => {
+    if (conceptSetOccurrences.length === 0) {
```
Comment (kemp-verily) [inline ui/src/datasets.tsx:136]: behaviour change?

File: ui/src/criteria/concept.tsx [hunk 2]
```diff
@@ -92,7 +92,7 @@ class _ implements CriteriaPlugin<Data> {
   }
 
   renderInline() {
-    return <ConceptInline data={this.data} />;
+    return <ConceptInline />;
```
Comment (kemp-verily) [inline ui/src/criteria/concept.tsx:95]: unrelated change?

**Review by kemp-verily — APPROVED** [review]
looks like a huge improvement!

File: ui/src/datasets.tsx (hunk 1 above)
Comment (tjennison-google) [reply ui/src/datasets.tsx:136]: This just fixes the default look of the checkbox when there are no concept sets selected vs. when all of the values of a concept set are unselected.

File: ui/src/criteria/concept.tsx (hunk 2 above)
Comment (tjennison-google) [reply ui/src/criteria/concept.tsx:95]: `props` became an unused variable with the removal of the concept in the inline display so I had to remove it entirely.

**Review by marikomedlock — APPROVED** [review]
Screenshots look really great.

File: ui/src/criteria/concept.tsx [hunk 3]
```diff
@@ -365,29 +374,12 @@ function ConceptEdit(props: ConceptEditProps) {
...
-    </>
+    </Box>
   );
 }
 
-type ConceptInlineProps = {
-  data: Data;
-};
-
-function ConceptInline(props: ConceptInlineProps) {
```
[the remark this answers is not in the document]
Comment (tjennison-google) [reply ui/src/criteria/concept.tsx:376]: Yes, though there's currently nothing to actually edit there. Eventually that's where modifiers will go. Once I removed the current content I ended up having to remove most of it entirely because it became a chain of unused variables.

File: ui/src/theme.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
+import { createTheme } from "@mui/material/styles";
+
+export const theme = createTheme({
```
[the remark this answers is not in the document]
Comment (tjennison-google) [reply ui/src/theme.tsx:3]: I think it will be an option for a javascript plugin that either wraps or replaces the argument to createTheme. Some high level things like logos and primary colors might make sense as part of the underlay or other config but full styling would require too much customization I think. I don't want to be recreating CSS in the underlay. Obviously anything related to the underlying data will still remain in the underlay though.
